lib/src/tasks/local/cli.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library dart_dev.src.tasks.local.cli;
+
+import 'dart:io';
+import 'dart:async';
+
+import 'package:args/args.dart';
+import 'package:path/path.dart';
+
+import 'package:dart_dev/util.dart' show reporter;
+import 'package:dart_dev/src/tasks/config.dart';
+
+import 'package:dart_dev/src/tasks/local/api.dart';
+import 'package:dart_dev/src/tasks/cli.dart';
+
+class Command {
+  final String path;
+  final String extension;
+  final String name;
+
+  Command(this.path, this.name, this.extension);
+}
+
+Directory _directory(String path) => new Directory(path);
+
+Function _parseCommand(String pattern) => (File f) {
```

Comment:
This isn't really a `_parseCommand`, right? It's really a function that returns a method that would be named that if it weren't anonymous. Maybe `_commandParser(...)`?